Gems/ROS2/Code/Source/Frame/ROS2FrameGameSystemComponent.cpp

```diff
@@ -0,0 +1,123 @@
+    {
+        if (AZ::SerializeContext* serialize = azrtti_cast<AZ::SerializeContext*>(context))
+        {
+            serialize->Class<ROS2FrameGameSystemComponent, AZ::Component>()->Version(1);
+        }
+    }
+
+    void ROS2FrameGameSystemComponent::Activate()
+    {
+    }
+
+    void ROS2FrameGameSystemComponent::Deactivate()
+    {
+        m_registeredEntities.clear();
+        m_entityIdToFrameId.clear();
+        m_frameIdToEntityId.clear();
+    }
+
+    void ROS2FrameGameSystemComponent::RegisterFrame(const AZ::EntityId& frameEntityId)
+    {
+        m_registeredEntities.insert(frameEntityId);
+
+        AZStd::string namespaceFrameId = "";
+        ROS2FrameComponentBus::EventResult(namespaceFrameId, frameEntityId, &ROS2FrameComponentBus::Events::GetNamespacedFrameID);
+        if (!namespaceFrameId.empty())
+        {
+            m_frameIdToEntityId[namespaceFrameId] = frameEntityId;
+            m_entityIdToFrameId[frameEntityId] = namespaceFrameId;
+        }
+    }
+
+    void ROS2FrameGameSystemComponent::UnregisterFrame(const AZ::EntityId& frameEntityId)
+    {
+        m_registeredEntities.erase(frameEntityId);
+        AZStd::string namespacedFrameId = m_entityIdToFrameId[frameEntityId];
+        m_frameIdToEntityId.erase(namespacedFrameId);
+        m_entityIdToFrameId.erase(frameEntityId);
+    }
+
+    const AZStd::unordered_set<AZ::EntityId>& ROS2FrameGameSystemComponent::GetRegisteredFrames() const
+    {
+        return m_registeredEntities;
+    }
+
+    bool ROS2FrameGameSystemComponent::IsFrameRegistered(const AZ::EntityId& frameEntityId) const
+    {
+        return m_registeredEntities.find(frameEntityId) != m_registeredEntities.end();
+    }
+
+    size_t ROS2FrameGameSystemComponent::GetRegisteredFrameCount() const
+    {
+        return m_registeredEntities.size();
+    }
+
+    AZStd::optional<AZ::EntityId> ROS2FrameGameSystemComponent::GetFrameEntityByNamespacedId(const AZStd::string& namespacedFrameId) const
+    {
+        auto it = m_frameIdToEntityId.find(namespacedFrameId);
+        return (it != m_frameIdToEntityId.end()) ? AZStd::optional<AZ::EntityId>(it->second) : AZStd::nullopt;
+    }
+
+    AZStd::optional<AZStd::string> ROS2FrameGameSystemComponent::GetNamespacedFrameId(const AZ::EntityId& frameEntityId) const
+    {
+        auto it = m_entityIdToFrameId.find(frameEntityId);
+        return (it != m_entityIdToFrameId.end()) ? AZStd::optional<AZStd::string>(it->second) : AZStd::nullopt;
+    }
+
+    AZStd::unordered_set<AZStd::string> ROS2FrameGameSystemComponent::GetAllNamespacedFrameIds() const
+    {
+        AZStd::unordered_set<AZStd::string> frameIds;
+        AZStd::ranges::transform(
+            m_entityIdToFrameId | AZStd::views::values,
+            AZStd::inserter(frameIds, frameIds.end()),
+            [](const auto& frameId)
+            {
+                return frameId;
+            });
+
+        return frameIds;
```

Comment:
Isn't it the same, but shorter?
```suggestion
        const auto valueView = AZStd::ranges::views::values(m_entityIdToFrameId);
        AZStd::unordered_set<AZStd::string> frameIds{ valueView.begin(), valueView.end() };

        return frameIds;
```